aptos-move/framework/aptos-framework/sources/coin.move

```diff
@@ -150,6 +187,167 @@ module aptos_framework::coin {
     /// Capability required to burn coins.
     struct BurnCapability<phantom CoinType> has copy, store {}
 
+    /// The mapping between coin and fungible asset.
+    struct CoinConversionMap has key {
+        coin_to_fungible_asset_map: Table<TypeInfo, address>,
+    }
+
+    #[resource_group_member(group = aptos_framework::object::ObjectGroup)]
+    /// The paired coin type info stored in fungible asset metadata object.
+    struct PairedCoinType has key {
+        type: TypeInfo,
+    }
+
+    inline fun borrow_conversion_map_mut(): &mut CoinConversionMap {
+        if (!exists<CoinConversionMap>(
+            @aptos_framework
+        ) && std::features::coin_to_fungible_asset_migration_feature_enabled()) {
+            move_to(&create_signer::create_signer(@aptos_framework), CoinConversionMap {
+                coin_to_fungible_asset_map: table::new<TypeInfo, address>(),
+            })
+        };
+        borrow_global_mut<CoinConversionMap>(@aptos_framework)
+    }
+
+    public entry fun pair_existing_fungible_asset<CoinType>(
+        coin_creator: &signer,
+        metadata: Object<Metadata>
+    ) acquires CoinConversionMap, CoinInfo {
+        // Verify the metadata is the same
+        assert!(
+            name<CoinType>() == fungible_asset::name(metadata),
```

Comment:
Names having to match seems like a potentially debatable requirement. Shouldn't this be up to the creator to decide?

Reply:
I think it is confusing otherwise? could you give an example?